File: java/claude/p19_nre.java
public class p19_nre {

    private static final int ADDEND = 1000;

    public static void main(String[] args) {
        /* check that an argument is provided */
        if (args.length != 1) {
            System.err.println("Usage: java p19_nre <integer>");
            System.exit(1);
        }

        long value;
        try {
            /* attempt safe integer parsing — Long.parseLong rejects non-integers and overflow */
            value = Long.parseLong(args[0]);
        } catch (NumberFormatException e) {
            /* if parsing fails → print an error and exit */
            System.err.println("Error: \"" + args[0] + "\" is not a valid integer.");
            System.exit(1);
            return;
        }

        /* before adding ADDEND, verify the result will not overflow long */
        if (value > Long.MAX_VALUE - ADDEND) {
            System.err.println("Error: " + value + " + " + ADDEND
                    + " would overflow the integer range.");
            System.exit(1);
        }
        if (value < Long.MIN_VALUE + ADDEND) {
            System.err.println("Error: " + value + " + " + ADDEND
                    + " would underflow the integer range.");
            System.exit(1);
        }

        /* if safe, compute and print the result */
        long result = value + ADDEND;
        System.out.println(result);
    }
}

File: python/claude/p19_nre.py
import sys

ADDEND = 1000

# Python's int is arbitrary-precision, so true overflow cannot occur.
# We guard against values that would overflow a 64-bit signed integer
# (C long / Java long) for cross-language consistency and portability.
INT64_MAX = (1 << 63) - 1
INT64_MIN = -(1 << 63)


def main() -> int:
    # check that an argument is provided
    if len(sys.argv) != 2:
        print(f"Usage: python {sys.argv[0]} <integer>", file=sys.stderr)
        return 1

    # attempt safe integer parsing
    try:
        value = int(sys.argv[1])
    except ValueError:
        # if parsing fails → print an error and exit
        print(f"Error: \"{sys.argv[1]}\" is not a valid integer.", file=sys.stderr)
        return 1

    # before adding ADDEND, verify the result will not overflow 64-bit signed range
    result = value + ADDEND
    if result > INT64_MAX:
        print(f"Error: {value} + {ADDEND} would overflow the 64-bit integer range.",
              file=sys.stderr)
        return 1
    if result < INT64_MIN:
        print(f"Error: {value} + {ADDEND} would underflow the 64-bit integer range.",
              file=sys.stderr)
        return 1

    # if safe, compute and print the result
    print(result)
    return 0


if __name__ == "__main__":
    sys.exit(main())
